x/gov/abci.go

```diff
@@ -173,6 +270,15 @@ func EndBlocker(ctx sdk.Context, keeper *keeper.Keeper) error {
 			endTime := proposal.VotingStartTime.Add(*params.VotingPeriod)
 			proposal.VotingEndTime = &endTime
 
+			if params.QuorumCheckCount > 0 {
+				// add proposal to quorum check queue
+				quorumTimeoutTime := proposal.VotingStartTime.Add(*params.QuorumTimeout)
+				err = keeper.QuorumCheckQueue.Set(ctx, collections.Join(quorumTimeoutTime, proposal.Id), v1.NewQuorumCheckQueueEntry(quorumTimeoutTime, params.QuorumCheckCount))
+				if err != nil {
+					return false, err
+				}
```

Comment:
This code is repeated in `keeper.ActivateVotingPeriod()`, we should create a dedicated `keeper.InitQuorumCheck()` or something like that.

Reply:
There's 2 lines, and the second one contains many instructions. If we outline everything we got:

```go
quorumTimeoutTime := proposal.VotingStartTime.Add(*params.QuorumTimeout)
key := collections.Join(quorumTimeoutTime, proposal.Id)
val := v1.NewQuorumCheckQueueEntry(quorumTimeoutTime, params.QuorumCheckCount)
err := keeper.QuorumCheckQueue.Set(ctx, val, key)
if err != nil {
	return false, err
}
```

Suppose that we decide to change the key or the value format, we may forget to update one or the other. For that reason I think keys and values format should be forged in a single location.

Side note: in general I think that having those queues field public is really a bad design. I would rather expose public methods on the `Keeper`, so then when we test `abci.EndBlocker()`, we can pass a mock of the `Keeper` and assert only method calls instead of complex store assertions.

Reply:
I still see it differently, to be honest with you. For me this is simple enough that does not require encapsulation, and is (and will be) used in just two places, so in my view it would be over-engineering.

With that said, if you feel so compelled to do this, I am not against it as it would not change anything in the structure or the logic. So feel free to maybe make these changes and make a PR to this branch